test/node/uuid_tests.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const { Buffer } = require('buffer');
+const BSON = require('../register-bson');
+const util = require('util');
+const { validate: uuidStringValidate, version: uuidStringVersion } = require('uuid');
+const UUID = BSON.UUID;
+
+// Test values
+const UPPERCASE_UUID_STRING = 'AAAAAAAA-AAAA-4AAA-AAAA-AAAAAAAAAAAA';
+const UPPERCASE_HEX_STRING = UPPERCASE_UUID_STRING.replace(/-/g, '');
+const LOWERCASE_UUID_STRING = UPPERCASE_UUID_STRING.toLowerCase();
+const LOWERCASE_HEX_STRING = LOWERCASE_UUID_STRING.replace(/-/g, '');
+
+describe('UUID', function () {
+  /**
+   * @ignore
+   */
+  it('should correctly generate a valid UUID v4 from empty constructor', function (done) {
```

Comment:
Isn't it possible to provide a function with zero arguments and then avoid having to call `done()`?

Reply:
Yup, but this is how it's written in all other tests... I'll change it.
`done` is to allow for asynchronous testing.